engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java

```diff
@@ -56,12 +60,20 @@ public WhereFilter getExpression(String expression, Matcher matcher, Object... a
                 }
                 log.debug().append("WhereFilterFactory creating MatchFilter for expression: ").append(expression)
                         .endl();
-                return new MatchFilter(MatchFilter.CaseSensitivity.MatchCase, columnName, matcher.group(2));
+                // @formatter:off
+                return new MatchFilter(
+                        MatchFilter.CaseSensitivity.MatchCase,
+                        matcher.group(3) == null // We must have matched EQUALS_PTRN, rather than NOT_EQUALS_PTRN
```

Comment:
this is very subtle

Reply:
That's why it has a comment. It's the same way as ICASE and NOT work.

Reply:
I re-did the whole class' handling of matcher groups, see if this is better.